freqtrade/persistence/models.py

```diff
@@ -860,6 +860,141 @@ def get_overall_performance(minutes=None) -> List[Dict[str, Any]]:
             for pair, profit, profit_abs, count in pair_rates
         ]
 
+    @staticmethod
+    def get_buy_tag_performance(pair: Optional[str]) -> List[Dict[str, Any]]:
+        """
+        Returns List of dicts containing all Trades, based on buy tag performance
+        Can either be average for all pairs or a specific pair provided
+        NOTE: Not supported in Backtesting.
+        """
+
+        if(pair is not None):
+            tag_perf = Trade.query.with_entities(
+                Trade.buy_tag,
+                func.sum(Trade.close_profit).label('profit_sum'),
+                func.sum(Trade.close_profit_abs).label('profit_sum_abs'),
+                func.count(Trade.pair).label('count')
+            ).filter(Trade.is_open.is_(False))\
+                .filter(Trade.pair == pair) \
+                .order_by(desc('profit_sum_abs')) \
+                .all()
+        else:
+            tag_perf = Trade.query.with_entities(
+                Trade.buy_tag,
+                func.sum(Trade.close_profit).label('profit_sum'),
+                func.sum(Trade.close_profit_abs).label('profit_sum_abs'),
+                func.count(Trade.pair).label('count')
+            ).filter(Trade.is_open.is_(False))\
+                .group_by(Trade.buy_tag) \
+                .order_by(desc('profit_sum_abs')) \
+                .all()
```

Comment:
> Also does anything need to be changed for btanalysis? I see it has a lot of column names, should "exit_tag" be included there?

i don't think so. used to convert database records to dataframes ... since we're storing the information in sell_reason - there's little/no reason to modify this.

Reply:
Looks good to me :) Thank you for your help!